include/translated-post.php

```diff
@@ -410,4 +410,76 @@ protected function get_db_infos() {
 			'default_alias' => $GLOBALS['wpdb']->posts,
 		);
 	}
+
+	/**
+	 * Wraps `wp_insert_post` with language feature.
+	 *
+	 * @since 3.7
+	 *
+	 * @param array        $postarr {
+	 *     Optional. An array of elements that make up a post to update or insert.
+	 *     All `wp_insert_post` parameters plus the following.
+	 *
+	 *     @type string[] $translations The translation group to assign to the post with language slug as keys and post ID as values.
+	 * }
+	 * @param PLL_Language $language The post language object.
+	 * @return int|WP_Error The post ID on success. The value `WP_Error` on failure.
+	 */
+	public function insert( array $postarr, PLL_Language $language ) {
+		$post_id = wp_insert_post( $postarr, true );
+		if ( is_wp_error( $post_id ) ) {
+			// Something went wrong!
+			return $post_id;
+		}
+
+		$this->set_language( $post_id, $language );
+
+		if ( ! empty( $postarr['translations'] ) ) {
+			$this->save_translations( $post_id, $postarr['translations'] );
+		}
+
+		return $post_id;
+	}
+
+	/**
+	 * Wraps `wp_update_post` with language feature.
+	 *
+	 * @since 3.7
+	 *
+	 * @param array $postarr {
+	 *     Optional. An array of elements that make up a post to update or insert.
+	 *     All `wp_update_post` parameters plus the following.
+	 *
+	 *     @type PLL_Language|string $lang         The post language object or slug.
+	 *     @type string[]            $translations The translation group to assign to the post with language slug as keys and post ID as values.
+	 * }
+	 * @return int|WP_Error The post ID on success. The value `WP_Error` on failure.
+	 */
+	public function update( array $postarr ) {
+		$post = get_post( $postarr['ID'] );
+		if ( ! $post instanceof WP_Post ) {
+			return new WP_Error( 'invalid_post', __( 'Empty Post.', 'polylang' ) );
+		}
+
+		if ( ! empty( $postarr['lang'] ) ) {
+			$language = $this->languages->get( $postarr['lang'] );
+			if ( ! $language instanceof PLL_Language ) {
```

Comment:
Avoid useless operation and align error message on WP:
```suggestion
		if ( ! empty( $postarr['lang'] ) ) {
			$post = get_post( $postarr['ID'] );
			if ( ! $post instanceof WP_Post ) {
				return new WP_Error( 'invalid_post', __( Invalid post ID.', 'polylang' ) );
			}

			$language = $this->languages->get( $postarr['lang'] );
			if ( ! $language instanceof PLL_Language ) {
```

Reply:
Why include the `WP_Post` check in the language checking condition?

Reply:
Because it will be done by WP. It's ok to check it if there is a new language as you are using `$postarr['ID']`. Otherwise it's double useless work.